Framework/Algorithms/src/SortXAxis2.cpp

```diff
@@ -44,23 +49,24 @@ void SortXAxis::exec() {
   MatrixWorkspace_sptr outputWorkspace = inputWorkspace->clone();
 
   // Check if it is a valid histogram here
-  bool isAProperHistogram = determineIfHistogramIsValid(inputWorkspace);
+  bool isAProperHistogram = determineIfHistogramIsValid(inputWorkspace.get());
```

Comment:
If we change to references then you can just use `*inputWorkspace` rather than needing to get the raw pointer at all.

Reply:
The getProperty function doesn't like outputting to a raw pointer hence my requirement to .get() the raw pointer.

Reply:
`getProperty` can only return `shared_ptr`. You already have `inputWorkspace` as a `shared_ptr` don't you here so you can just use `determineIfHistogramIsValid(*inputWorkspace);` instead?

Reply:
I can yes! Updated the code accordingly.